src/elements/EnvironmentVariables/AddEnvironmentVariablesRow.js

```diff
@@ -0,0 +1,161 @@
+import PropTypes from 'prop-types'
+
+import Input from '../../common/Input/Input'
+import Select from '../../common/Select/Select'
+import Tooltip from '../../common/Tooltip/Tooltip'
+import TextTooltipTemplate from '../TooltipTemplate/TextTooltipTemplate'
+
+import { ReactComponent as Plus } from '../../images/plus.svg'
+import { ReactComponent as Delete } from '../../images/delete.svg'
+import {
+  isNameNotUnique,
+  SECRET_TYPE,
+  secretKeyValidationTip,
+  secretNameValidationTip,
+  selectTypeOptions
+} from './environmentVariables.util'
+
+const AddEnvironmentVariablesRow = ({
+  addVariable,
+  discardChanges,
+  newVariable,
+  setNewVariable,
+  setValidation,
+  validation,
+  variables
+}) => {
+  return (
+    <div className="table__body">
+      <div className="table__body-column">
+        <div className="input-row-wrapper">
+          <Input
+            className="input-row__item"
+            floatingLabel
+            invalid={
+              isNameNotUnique(newVariable.name, variables) ||
+              !validation.isNameValid
+            }
+            invalidText={
+              isNameNotUnique(newVariable.name, variables)
+                ? 'Name already exists'
+                : 'This field is invalid'
+            }
+            label="Name"
+            onChange={name => setNewVariable(state => ({ ...state, name }))}
+            required
+            setInvalid={value =>
+              setValidation(state => ({
+                ...state,
+                isNameValid: value
+              }))
+            }
+          />
+          <Select
+            onClick={type => {
+              setNewVariable(state => ({
+                ...state,
+                type,
+                value: '',
+                secretName: '',
+                secretKey: ''
+              }))
+              setValidation(prevState => ({
+                ...prevState,
+                isNameValid: true,
+                isValueValid: true,
+                isSecretKeyValid: true,
+                isSecretNameValid: true
+              }))
+            }}
+            options={selectTypeOptions}
+            selectedId={newVariable.type}
+          />
+          {newVariable.type === SECRET_TYPE ? (
+            <div className="input-row__item-secret">
+              <Input
+                className="secret-item"
+                floatingLabel
+                invalid={!validation.isSecretNameValid}
+                label="Secret Name"
+                onChange={secretName =>
+                  setNewVariable(state => ({ ...state, secretName }))
+                }
+                pattern="^(?=[\S\s]{1,253}$)[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]{0,61}[a-z0-9])?)*$"
+                tip={secretNameValidationTip}
+                setInvalid={value =>
+                  setValidation(state => ({
+                    ...state,
+                    isSecretNameValid: value,
+                    isValueValid: true
```

Comment:
no need to set validation for the value here, you already set it to true when you switch the type of the env